### ISSUE TITLE
Sockets are not closed properly in 1.6

### ISSUE THREAD

**Issue by ajardan** [containous/traefik#3263]
### Do you want to request a *feature* or report a *bug*?
bug
### What did you do?

Deployed traefik 1.6 with docker backend into our swarm cluster

### What did you expect to see?

Stable traefik as it was with the previous version

### What did you see instead?

The number of used sockets started increasing, and I have to kill all traefik containers about once a day (see screenshot)
![screenshot 2018-04-27 11 31 45_preview](https://user-images.githubusercontent.com/347796/39573658-6726e87c-4ed4-11e8-8b0e-265397c4ab66.png)

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      v1.6.0-rc6
Codename:     tetedemoine
Go version:   go1.10.1
Built:        2018-04-17_11:59:31AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
/traefik -c /dev/null --docker --docker.watch --docker.swarmmode --docker.domain=tld.com --api --metrics --metrics.prometheus '--entryPoints=Name:traefik Address::9080' '--entryPoints=Name:http Address::80' '--entryPoints=Name:https Address::443 TLS:/certs/tld.com.pem,/keys/tld.com.key' -l WARNING --respondingtimeouts.idletimeout=2s --maxidleconnsperhost=50 --forwardingtimeouts.dialtimeout=5s --forwardingtimeouts.responseheadertimeout=10s --ping
```

**Comment by ajardan** [comment]
Some more info:
```
# lsof -nP | grep sock | grep -c traefik
2059193
```

And a few records from lsof output:

```
traefik    1298                    root  cwd       DIR               0,41        58  126318491 /
traefik    1298                    root  rtd       DIR               0,41        58  126318491 /
traefik    1298                    root  txt       REG                9,2  51558016 2148000871 /traefik
traefik    1298                    root    0r     FIFO                0,8       0t0  126327836 pipe
traefik    1298                    root    1w     FIFO                0,8       0t0  126327837 pipe
traefik    1298                    root    2w     FIFO                0,8       0t0  126327838 pipe
traefik    1298                    root    3u     sock                0,7       0t0  126093706 protocol: TCP
traefik    1298                    root    4u  a_inode                0,9         0       7104 [eventpoll]
traefik    1298                    root    5u     sock                0,7       0t0  126093707 protocol: TCP
traefik    1298                    root    6u     sock                0,7       0t0  126093708 protocol: TCP
traefik    1298                    root    7u     sock                0,7       0t0  126093709 protocol: UNIX
traefik    1298                    root    8u     sock                0,7       0t0  126346170 protocol: TCP
traefik    1298                    root    9u     sock                0,7       0t0  126354358 protocol: TCP
traefik    1298                    root   10u     sock                0,7       0t0  126408102 protocol: TCP
traefik    1298                    root   11u     sock                0,7       0t0  126482901 protocol: TCP
traefik    1298                    root   12u     sock                0,7       0t0  126372440 protocol: TCP
traefik    1298                    root   13u     sock                0,7       0t0  126640796 protocol: TCP
traefik    1298                    root   14u     sock                0,7       0t0  126370699 protocol: TCP
traefik    1298                    root   15u     sock                0,7       0t0  126361596 protocol: TCP
traefik    1298                    root   16u     sock                0,7       0t0  126551698 protocol: TCP
traefik    1298                    root   17u     sock                0,7       0t0  126506309 protocol: TCP
traefik    1298                    root   18u     sock                0,7       0t0  126414792 protocol: TCP
traefik    1298                    root   19u     sock                0,7       0t0  126726068 protocol: TCP
traefik    1298                    root   20u     sock                0,7       0t0  126379784 protocol: TCP
traefik    1298                    root   21u     sock                0,7       0t0  126630988 protocol: TCP
traefik    1298                    root   22u     sock                0,7       0t0  127215538 protocol: TCP
traefik    1298                    root   23u     sock                0,7       0t0  126328458 protocol: TCP
traefik    1298                    root   24u     sock                0,7       0t0  126763828 protocol: TCP
traefik    1298                    root   25u     sock                0,7       0t0  126332223 protocol: TCP
traefik    1298                    root   26u     sock                0,7       0t0  126449527 protocol: TCP
traefik    1298                    root   27u     sock                0,7       0t0  126572712 protocol: TCP
traefik    1298                    root   28u     sock                0,7       0t0  126420813 protocol: TCP
```

**Comment by Juliens** [comment]
I'm sorry but I can't reproduce it. I have tried a simple configuration with one "whoami" container and I try to make lots of request.

Could you provide us more information about your stack?

**Comment by ajardan** [comment]
@Juliens sure, I would be happy any information that is required to fix this.

We deploy traefik as a docker swarm service in global mode. Ports 443 and 80 are esposed in "host" mode, so traefik receives the requests as it would run on the actual host. We receive requests from a high number of clients, most of the time new requests.

What other details would help to debug this ?

**Comment by ajardan** [comment]
I should probably also mention a lot of these are https connections terminated by traefik using a certificate mounted from the host machine into the traefik container. Here is a graph (traefik was killed at the end of graph, that's why there is a drop)

<img width="1363" alt="screenshot 2018-05-09 15 38 36" src="https://user-images.githubusercontent.com/347796/39817718-3a7a5e06-539f-11e8-9975-57eb15fb21a2.png">

**Comment by Juliens** [comment]
What do you mean by "https connections terminated by traefik"?

**Comment by ajardan** [comment]
I mean the actual HTTPS connections reach traefik directly, where the certificates are installed as well.

**Comment by ajardan** [comment]
I also see that memory usage is growing together with the number of open sockers (which is probably expected)

<img width="717" alt="screen shot 2018-05-17 at 17 15 08" src="https://user-images.githubusercontent.com/347796/40186737-1ffb22a6-59f6-11e8-8ed1-e4a7bb5286b5.png">

**Comment by Juliens** [comment]
Could you try to do this https://github.com/containous/traefik/issues/2673#issuecomment-373022820 and send us all the data you collect?

**Comment by ajardan** [comment]
Where could I send it privately, since this is a production balancer, so it may contain data not to be made public?

**Comment by Juliens** [comment]
You can join our [Slack workspace](https://traefik.herokuapp.com/) and send me a private message?

**Comment by Juliens** [comment]
After some research, I was unable to reproduce this problem.

There seems to be something with a client, but I can not find what and what kind of client. If you have information that can help, I would be happy to continue to research this problem.

In addition, I think it's the same bug as # 3165.

**Comment by ajardan** [comment]
I was able to reproduce the issue using these scripts even in 1.7

https://github.com/ajardan/socket-test

**Comment by C-Duv** [comment]
I might be running in the same issue, if not:

At first, only few hours after starting my Docker (via `docker-compose`: not swarm, not cluster, not Kubernetes) Træfik container (official *traefik* v1.6.5 -11569c001780- image) it would hang (HTTP clients would get timeout).
Log shown it was complaining about "too many open files":

> Accept error: accept tcp [::]:80: accept4: too many open files; retrying in 5ms

And indeed Træfik process was hitting the default 1024 files as the following command highlighted:

`ls -1 "/proc/$(ps -aef | grep '[0-9] /traefik' | awk '{ print $2 } ')/fd" | wc -l`

I restarted the container with higher limit:

```Yaml
services:
  reverse-proxy:
    image: traefik
    [...]
    ulimits:
      nofile:
        soft: 65000
        hard: 65000
        hard: 65000
```

And then could run fine for hours.
But I kept watching the number of open files and quickly noticed it keeps increasing: it's now 10481 (after 6 days and 216,100 requests served).

_ajardan_'s command `lsof -nP | grep sock | grep -c traefik` returns 293202.

I mainly have a HTTP endpoint, the HTTPS endpoint is for Træfik's dashboad itself.

**Comment by traefiker** [comment]
Closed by #3740.

**Comment by C-Duv** [comment]
Version 1.6.6 / #3740 resolved my _Accept error: accept tcp [::]:80: accept4: too many open files; retrying in 5ms_ issue.
It's been ~24 hours and the number of open files is still around ~21 files (while it was around 2230 after 24h on v1.6.5).
